util/error_document.go

```diff
@@ -20,3 +22,10 @@ func ErrorDoc(message, status string) *ErrorDocument {
 		}},
 	}
 }
+
+type NotFoundError struct {
+}
+
+func (e *NotFoundError) Error() string {
+	return "record not found"
+}
```

Comment:
So right here I would flesh this struct out a bit more, here are my suggestions:

1. Change the name to `ErrNotFound` to follow Go conventions (example here: https://pkg.go.dev/io/fs#ErrPermission)
2. Add an `Is` method so that `errors.Is` will check more than just the type + string, so we can make the string whatever we want and be dynamically generated when we call `Error()`